### PR TITLE
Updating in Mapping

### PR DESCRIPTION
**Pull request by VictorH2208**
This PR updates the mapping functions of both credential and presentation. To see how it works, can run the unit test at the end of the each files. Or run `cargo test test_to_anoncreds_presentation` for the presentation procedure or `cargo test test_to_w3c` for credentials. The output files can be found in the "samples" folder. The file "w3c_presentation" is the output of converting from AnonCreds to W3C, the "anoncreds_output" is the output of converting back to AnonCreds, and the "anoncreds_presentation_list" is the input that contains the presentation proof, schema and nonce used.

### CODE REVIEW

**Comment by swcurran** [comment]
@mikelodder7 — can you please take a look? This was discussed at the 2024025 AnonCreds Working Group meeting — [recording here](https://youtu.be/CgOi8bw5dzs) — and there are some questions for you.

@VictorH2208 — can you please take a look at addressing the lint issues in the failed test above?  It looks like a lot, but I think most of them will be easy to address.

**Comment by VictorH2208** [comment]
> @mikelodder7 — can you please take a look? This was discussed at the 2024025 AnonCreds Working Group meeting — [recording here](https://youtu.be/CgOi8bw5dzs) — and there are some questions for you.
> 
> @VictorH2208 — can you please take a look at addressing the lint issues in the failed test above? It looks like a lot, but I think most of them will be easy to address.

Yep. Looking at them right now.

**Review by mikelodder7 — APPROVED** [review]
Can we add an explainer to this? For future PR. This is hard to follow without one and there’s no explanation for the constants about where they come from and why they’re needed.
